lib/devicetrust/authn/authn_test.go

```diff
@@ -71,37 +88,35 @@ func TestRunCeremony(t *testing.T) {
 }
 
 func resetNative() func() {
-	gdc := authn.GetDeviceCredential
-	cdd := authn.CollectDeviceData
-	sc := authn.SignChallenge
+	getDeviceCredential := authn.GetDeviceCredential
+	collectDeviceData := authn.CollectDeviceData
+	signChallenge := authn.SignChallenge
+	getDeviceOSType := authn.GetDeviceOSType
+	solveTPMAuthDeviceChallenge := authn.SolveTPMAuthDeviceChallenge
 	return func() {
-		authn.GetDeviceCredential = gdc
-		authn.CollectDeviceData = cdd
-		authn.SignChallenge = sc
+		authn.GetDeviceCredential = getDeviceCredential
+		authn.CollectDeviceData = collectDeviceData
+		authn.SignChallenge = signChallenge
+		authn.GetDeviceOSType = getDeviceOSType
+		authn.SolveTPMAuthDeviceChallenge = solveTPMAuthDeviceChallenge
 	}
 }
 
-func enrollDevice(ctx context.Context, devices devicepb.DeviceTrustServiceClient, dev *testenv.FakeMacOSDevice) error {
+func enrollDevice(ctx context.Context, devices devicepb.DeviceTrustServiceClient, dev testenv.FakeDevice) error {
```

Comment:
@codingllama what do you think about us just calling `enroll.RunCeremony` instead? This is essentially becoming a like-for-like re-implementation of `enroll.RunCeremony` - whilst there's some concerns about a potential cyclical dependency - perhaps that's a sign we should just have the contents of both the `enroll` and `authn` packages in the `lib/devicetrust` package directly ?

Reply:
If we can cleanly swap the "native" bits for the fake devices, then I see no issue. Let's pull the refactor into another PR, though.

Also relevant: https://github.com/gravitational/teleport/pull/27039.

Reply:
> perhaps that's a sign we should just have the contents of both the enroll and authn packages in the lib/devicetrust package directly

This part I didn't follow, why would we need that?

Reply:
> If we can cleanly swap the "native" bits for the fake devices, then I see no issue. Let's pull the refactor into another PR, though.

Agreed on a seperate PR

> This part I didn't follow, why would we need that?

I was concerned that the `authn_test`  package having a dependency on the `enroll` package would introduce the likelihood of a circular dependency between the packages in future. In their current state, this wouldn't be the case though.

Reply:
It should be alright, one of the advantages of _test packages is that they are separate packages for the purposes of dependency cycles.